model/src/main/kotlin/LicenseFinding.kt

```diff
@@ -23,6 +23,7 @@ import com.fasterxml.jackson.annotation.JsonInclude
 
 import org.ossreviewtoolkit.model.config.LicenseFindingCuration
 import org.ossreviewtoolkit.utils.spdx.SpdxExpression
+import org.ossreviewtoolkit.utils.spdx.isSpdxExpression
```

Comment:
Commit message nits:
- "Add *an* option ..."
- "Scanners might return licenses that cannot be parsed as SPDX license expressions."
- "Provide a configuration option for user to map ..."

> This functionality is not limited to invalid SPDX expressions. The user
can also map unwanted licenses, like `LicenseRef-scancode-unknown` to
`NOASSERTION`.

I believe this paragraph is misleading, as it sounds as if mapping to `NOASSERTION` would get rid of the detected license (similar to concluding a declared license to `NONE`), which is not the case.

- "... ORT has to handle only valid SPDX expressions internally."

> The disadvantage of this approach is, that mapped licenses are stored in the
ScanResultStorage

Hmm, wasn't one point of doing this to *not* store the mapped licenses, so no rescan is required when the mapping changes?

Reply:
> > The disadvantage of this approach is, that mapped licenses are stored in the
> > ScanResultStorage
> 
> Hmm, wasn't one point of doing this to _not_ store the mapped licenses, so no rescan is required when the mapping changes?

This was my / our initial plan (See: https://github.com/oss-review-toolkit/ort/issues/5345#issuecomment-1125130550). However, implementing this would require to basically remove the `.toSpdx()` calls everywhere before storing anything in the ScanStorage and only handle Strings within ORT.
E.g. `BSD (3-clause)` is not a valid SPDX expression and even uses characters which are reserved in SPDX: `(` and `)` for indicating order of operations in license choices.

That's why I chose to implement it differently.